src/OSPSuite.Core/Domain/Formulas/DistributionFormulaFactory.cs

```diff
@@ -21,68 +25,85 @@ public interface IDistributionFormulaFactory
-         return distribution;
-      }
+         addObjectPathToDistributionFormula(distributionFormula, meanParameterName, MEAN, dimension);
+         addObjectPathToDistributionFormula(distributionFormula, deviationParameter, GEOMETRIC_DEVIATION, null);
+      };
+
+      public UniformDistributionFormula CreateUniformDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter minParameter, IParameter maxParameter) =>
+         createDistribution<UniformDistributionFormula>(minParameter.Dimension, configureUniformDistributionDistribution(minParameter.Name, maxParameter.Name));
 
-      public LogNormalDistributionFormula CreateLogNormalDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter meanParameter, IParameter deviationParameter)
+      private Action<DistributionFormula, IDimension> configureUniformDistributionDistribution(string minParameterName = null, string maxParameterName = null) => (distributionFormula, dimension) =>
       {
-         var distribution = createDistribution<LogNormalDistributionFormula>();
-         distribution.Dimension = meanParameter.Dimension;
-         addObjectPathToDistributionFormula(distribution, distributedParameter,meanParameter,Constants.Distribution.MEAN);
-         addObjectPathToDistributionFormula(distribution, distributedParameter, deviationParameter, Constants.Distribution.GEOMETRIC_DEVIATION);
-         return distribution;
-      }
+         addObjectPathToDistributionFormula(distributionFormula, minParameterName, MINIMUM, dimension);
+         addObjectPathToDistributionFormula(distributionFormula, maxParameterName, MAXIMUM, dimension);
+      };
 
-      public UniformDistributionFormula CreateUniformDistributionFormulaFor(IDistributedParameter distributedParameter, IParameter minParameter, IParameter maxParameter)
+      public DistributionFormula CreateFor(DistributionType distributionType, IDimension dimension)
       {
-         var distribution = createDistribution<UniformDistributionFormula>();
-         distribution.Dimension = minParameter.Dimension;
-         addObjectPathToDistributionFormula(distribution, distributedParameter,minParameter,Constants.Distribution.MINIMUM);
-         addObjectPathToDistributionFormula(distribution, distributedParameter,maxParameter,Constants.Distribution.MAXIMUM);
-         return distribution;
+         switch (distributionType)
+         {
+            case DistributionType.Normal:
+               return createDistribution<NormalDistributionFormula>(dimension, configureNormalDistribution());
+            case DistributionType.LogNormal:
+               return createDistribution<LogNormalDistributionFormula>(dimension, configureLogNormalDistributionDistribution());
+            case DistributionType.Uniform:
+               return createDistribution<UniformDistributionFormula>(dimension, configureLogNormalDistributionDistribution());
```

Comment:
Must be `configureUniformDistributionDistribution`

Reply:
yep. That's why some tests were failing :)